x/ccv/provider/keeper/relay.go

```diff
@@ -222,7 +222,23 @@ func (k Keeper) QueueVSCPackets(ctx sdk.Context) {
 
 	for _, chain := range k.GetAllConsumerChains(ctx) {
 		currentValidators := k.GetConsumerValSet(ctx, chain.ChainId)
-		nextValidators := k.ComputeNextEpochConsumerValSet(ctx, chain.ChainId, bondedValidators)
+		var nextValidators []providertypes.ConsumerValidator
+
+		considerOnlyOptIn := func(validator stakingtypes.Validator) bool {
```

Comment:
code duplicated from proposal.go, could we just extract a function here and put it into partial_set_security.go?

Reply:
Good point. I created `ShouldConsiderOnlyOptIn` (and added a test for it) and now do:
```
nextValidators := k.ComputeNextEpochConsumerValSet(ctx, chain.ChainId, bondedValidators,
	func(validator stakingtypes.Validator) bool {
		return k.ShouldConsiderOnlyOptIn(ctx, chain.ChainId, validator)
	})
```